app/apollo/resolvers/subscription.js

```diff
@@ -739,24 +802,34 @@ const subscriptionResolvers = {
       const user = whoIs(me);
 
       try {
-        logger.info( {req_id, user, org_id, uuid }, `${queryName} validating` );
+        logger.info( {req_id, user, org_id, uuid}, `${queryName} validating`);
 
         validateString( 'org_id', org_id );
         validateString( 'uuid', uuid );
 
-        //var subscription = await models.Subscription.findOne({ org_id, uuid });
         const conditions = await getGroupConditionsIncludingEmpty(me, org_id, ACTIONS.READ, 'name', queryName, context);
-        logger.debug({req_id, user, org_id, conditions }, `${queryName} group conditions are...`);
-        var subscription = await models.Subscription.findOne({ org_id, uuid, ...conditions }, {});
+        logger.debug({req_id, user, org_id, conditions}, `${queryName} group conditions are...`);
 
-        if(!subscription){
-          throw new NotFoundError(context.req.t('Subscription uuid "{{uuid}}" not found.', {'uuid':uuid}), context);
-        }
+        // Find the subscription
+        let subscription = await models.Subscription.findOne({ org_id, uuid, ...conditions }, {});
+        logger.info({req_id, user, org_id, uuid}, `${queryName} validating - found: ${!!subscription}`);
 
-        await validAuth(me, org_id, ACTIONS.DELETE, TYPES.SUBSCRIPTION, queryName, context, [subscription.uuid, subscription.name]);
+        const subscriptionIdentifiers = subscription ? [uuid, subscription.name] : [uuid];
+        await validAuth(me, org_id, ACTIONS.DELETE, TYPES.SUBSCRIPTION, queryName, context, subscriptionIdentifiers);
+        logger.info( {req_id, user, org_id, uuid}, `${queryName} validating - authorized` );
 
-        // loads the channel
-        var channel = await models.Channel.findOne({ org_id, uuid: subscription.channel_uuid });
+        // Find the channel
+        const channel = await models.Channel.findOne({ org_id, uuid: subscription.channel_uuid });
+        logger.info({req_id, user, org_id, uuid}, `${queryName} validating - found: ${!!channel}`);
+
+        const channelIdentifiers = channel ? [subscription.channel_uuid, channel.name] : [subscription.channel_uuid];
+        await validAuth(me, org_id, ACTIONS.READ, TYPES.CHANNEL, queryName, context, channelIdentifiers);
+        logger.info({req_id, user, org_id, uuid}, `${queryName} validating - channel authorized`);
+
+        // Check for errors after validation
+        if (!subscription) {
+          throw new NotFoundError(context.req.t('Subscription uuid "{{uuid}}" not found.', {'uuid':uuid}), context);
+        }
```

Comment:
- this check is _part_ of validation, not "after validation".
- The subscription check should be moved up to co-locate with where the subscription is retrieved and auth verified.  If the subscription is not found, there's no reason to do the expensive channel lookup and auth check.

Reply:
Resolved